### DISCUSSION TITLE
Panicing when running on my crate

### DISCUSSION THREAD

**Discussion by zxch3n** [SeaQL/FireDBG.for.Rust#12] [Bug]
I attempted to run the examples in my crate [`loro`](https://github.com/loro-dev/loro) and encountered an issue. 

```
warning: list was compiled with optimization - stepping may behave oddly; variables may not be available.
thread 'main' panicked at debugger/src/debugger.rs:386:21:
assertion failed: sb_frame.sp() <= active_frame.stack_pointer
```

The persistent warning, `warning: list was compiled with optimization - stepping may behave oddly; variables may not be available`, appears regardless of my configuration attempts. I am uncertain if this is the cause of the issue.

**Comment by zxch3n** [comment]
The program runs for a while after I remove all `#[inline(always)]`. Then:

```
thread 'main' panicked at debugger/src/debugger.rs:674:17:
assertion failed: sb_frame.sp() >= last_frame.stack_pointer
```

**Reply by tyt2y3** [reply]
Yes, you're right, inlining can be problematic. I'd suggest you to change your code to `#[cfg_attr(not(debug_assertions), inline(always))]`. That way you don't inline on debug build, and inline on production.

**Reply by zxch3n** [reply]
Understood. Is it possible to resolve this issue within the current project, or do we have other configuration options to circumvent the problem?

**Reply by tyt2y3** [reply]
Yes, I think so. On our side when parsing the source files, we can probably ignore `#[inline]` functions.
